src/core/Akka.TestKit/TestKitBase_Receive.cs

```diff
@@ -383,20 +395,34 @@ public ValueTask<(bool success, MessageEnvelope envelope)> TryPeekOneAsync(TimeS
             if (maxDuration.IsZero())
             {
                 ConditionalLog(shouldLog, "Trying to peek message from TestActor queue. Will not wait.");
-                var didPeek = _testState.Queue.TryPeek(out var item);
+                var didPeek = _testState.Queue.Reader.TryPeek(out var item);
                 peek = (didPeek, item);
             }
             else if (maxDuration.IsPositiveFinite())
             {
                 ConditionalLog(shouldLog, "Trying to peek message from TestActor queue within {0}", maxDuration);
-                peek = await _testState.Queue.TryPeekAsync((int)maxDuration.TotalMilliseconds, cancellationToken)
-                    ;
+                var delayTask = Task.Delay(maxDuration, cancellationToken);
```

Comment:
Same as above, possible memory leak

Reply:
Not the case, as I explained in my earlier comment